android/app/src/main/java/org/fossasia/openevent/utils/ISO8601Date.java

```diff
@@ -3,135 +3,216 @@
+        String end = ISO8601Date.getTimeZoneDateStringFromString(endDate);
+        String combinedDateString = ISO8601Date.getDateStringSplit(start, 0) + "," + ISO8601Date.getDateStringSplit(start, 1) + " - " +
+                ISO8601Date.getDateStringSplit(end, 0) + "," + ISO8601Date.getDateStringSplit(end, 1);
+        return combinedDateString;
     }
 
-    public static String get12HourTime(final Date date) {
+    //returns day and date of the event
+    public static String getDateFromStartDateString(String startDate) {
+        String start = ISO8601Date.getTimeZoneDateStringFromString(startDate);
+        String combinedDateString = ISO8601Date.getDateStringSplit(start, 0) + "," + ISO8601Date.getDateStringSplit(start, 1);
+        return combinedDateString;
+    }
 
-        SimpleDateFormat dateFormat = new SimpleDateFormat("hh:mm a", Locale.getDefault());
-        dateFormat.setTimeZone(getTimezone());
-        return dateFormat.format(date);
+    //return combined time and timezone string from start string
+    public static String getTimeFromStartDateString(String startDate) {
+        String start = ISO8601Date.getTimeZoneDateStringFromString(startDate);
+        String combinedDateString = ISO8601Date.getDateStringSplit(start, 2) + "," + ISO8601Date.getDateStringSplit(start, 3);
+        return combinedDateString;
     }
 
-    public static  String getDate(final Date date) {
-       SimpleDateFormat dateFormat = new SimpleDateFormat("EEE, MMM d, yyyy", Locale.getDefault());
-        dateFormat.setTimeZone(getTimezone());
-        return dateFormat.format(date);
+    //return combined time and timezone string from end string
+    public static String getTimeFromEndDateString(String endDate) {
+        String end = ISO8601Date.getTimeZoneDateStringFromString(endDate);
+        String combinedDateString = ISO8601Date.getDateStringSplit(end, 2) + "," + ISO8601Date.getDateStringSplit(end, 3);
+        return combinedDateString;
     }
 
+    //return combined 12 hour date string from start and end string
+    public static String get12HourTimeFromCombinedDateString(String startDate, String endDate) {
+        String startString = ISO8601Date.get12HourTimeFromString(startDate);
+        String endString = ISO8601Date.get12HourTimeFromString(endDate);
+        return startString + " - " + endString;
+    }
 
-    public static Date getDateObject(final String iso8601String) {
-        setEventTimezone();
-        String final1 = iso8601String + "Z";
+    //return date string of a specified timezone from a given date string
+    public static String getTimeZoneDateStringFromString(String dateString) {
+        String currentDate = dateString + "Z";
+        SimpleDateFormat simpleDateFormat = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'", Locale.getDefault());
+        simpleDateFormat.setTimeZone(TimeZone.getDefault());
+        Date date = null;
+        String date_string = null;
+        try {
+            if(dateString!=null){
```

Comment:
please insert a space before the starting curly braces in `if` statement. Please do so in every subsequent place possible.